Prepare release of CumulusCI 3.34 (#2592)
SFDO-Tooling/CumulusCI: HISTORY.rst

History
=======

> 3.34 (2022-04-29)

Reviewer: ```suggestion
3.34.0 (2021-04-29)
```

-----------------

Critical Changes:

- If you have custom flows that utilize the ``github_release`` task, they will need to be updated to include the ``package_type`` option (which is required). (#2546)

Changes:

- The ``github_release`` task now has a ``package_type`` option which is included in the information written to GitHub release tags. The following standard library "release" flows have been updated with hardcoded values (either ``1GP`` or ``2GP``) for this option:
    - ``release_beta`` (1GP)
    - ``release_production`` (1GP)
    - ``release_2gp_beta`` (2GP) 
    - ``release_2gp_production`` (2GP)

> (#2546)

Reviewer: ```suggestion
  (#2546)
```

- The ``update_dependencies`` task now supports a ``packages_only`` option, which suppresses the installation of unpackaged metadata dependencies. This option is intended to support building update-only or idempotent installers. (#2587)

- The ``github_automerge_main`` task has a new option, ``skip_future_releases``, which can be set to ``False`` to disable the default behavior of skipping branches that are numeric (and thus considered release branches) but not the lowest number. (#2582)

> - Added an new option ``set_recently_viewed`` to the load_dataset task that sets newly inserted data as recently viewed. This changes the default behavior.  By default (if you do not specify the option), the first 1000 records inserted via the Bulk API will be set as recently viewed. If fewer than 1000 records are inserted, existing objects of the same type being inserted will also be set as recently viewed. (#2578)

Reviewer: ```suggestion
- Added an new option ``set_recently_viewed`` to the ``load_dataset`` task that sets newly inserted data as recently viewed. This changes the default behavior.  By default (if you do not specify the option), the first 1000 records inserted via the Bulk API will be set as recently viewed. If fewer than 1000 records are inserted, existing objects of the same type being inserted will also be set as recently viewed. (#2578)
```

> - ``update_dependencies`` can now consume 2GP releases in upstream repositories, provided they're stored in release tags as generated by CumulusCI. (#2557)

Reviewer: ```suggestion
- The ``update_dependencies`` task can now consume 2GP releases in upstream repositories, provided they're stored in release tags as generated by CumulusCI. (#2557)
```

> - Bulk Data tasks now support adding or removing a namespace from a mapping file to match the target org for sObjects and not just fields. (#2532)

Reviewer: ```suggestion
- The ``extract_dataset`` and ``load_datast`` tasks now support adding or removing a namespace from a mapping file to match the target org for sObjects and not just fields. (#2532)
```

- The ``create_package_version`` task can now increment package version numbers when the package is not in a released state. (#2547)

Issues Closed

- Fixed an error in the ``github_automerge_main`` task when using a branch prefix that doesn't contain a slash. (#2582)

> - Fixed SOQL query for ``push_api`` library's ``get_newer_released_version_objs`` and ``get_older_released_version_objs`` to allow for proper package version queries. (#2577)

Reviewer: ```suggestion
- Fixed logic in the ``push_sandbox`` and ``push_all`` tasks which was selecting the wrong package versions. (#2577)
```

- Improved logging of errors from sfdx while converting sfdx format metadata to deploy via the Metadata API, so that they are not lost when CumulusCI is embedded in another system like MetaCI or Metecho. (#2574)

> - Includes Snowfakery 1.10 with upgrades to its Fake data functions. (https://github.com/SFDO-Tooling/Snowfakery/releases/tag/v1.10)

Reviewer: This belongs in Changes, not Issues Closed
